src/Stratis.Features.FederatedPeg/Wallet/FederationWalletManager.cs

```diff
@@ -809,23 +808,34 @@ public IEnumerable<IWithdrawal> GetWithdrawals()
         }
 
         /// <inheritdoc />
-        public List<(Transaction, TransactionData, IWithdrawal)> FindWithdrawalTransactions(uint256 depositId = null)
+        public List<(Transaction, IWithdrawal)> FindWithdrawalTransactions(uint256 depositId = null)
         {
+            // A withdrawal is a transaction that spends funds from the multisig wallet.
             lock (this.lockObject)
             {
-                List<(Transaction, TransactionData, IWithdrawal)> withdrawals = new List<(Transaction, TransactionData, IWithdrawal)>();
+                var withdrawals = new List<(Transaction, IWithdrawal)>();
 
                 foreach (TransactionData transactionData in this.Wallet.MultiSigAddress.Transactions)
                 {
-                    Transaction walletTran = transactionData.GetFullTransaction(this.network);
-                    IWithdrawal withdrawal = this.withdrawalExtractor.ExtractWithdrawalFromTransaction(walletTran, transactionData.BlockHash, transactionData.BlockHeight ?? 0);
+                    if (transactionData.SpendingDetails == null)
+                        continue;
+
+                    Transaction walletTran = this.network.CreateTransaction(transactionData.SpendingDetails.Hex);
+
+                    if (withdrawals.Any(w => w.Item1.GetHash() == walletTran.GetHash()))
+                        continue;
+
+                    int? blockHeight = transactionData.SpendingDetails.BlockHeight;
+                    uint256 blockHash = (blockHeight == null) ? null : this.chainIndexer.GetHeader((int)blockHeight).HashBlock;
```

Comment:
I think there is something wrong with the logic here.

If block height is not null we get the block from the ChainIndexer otherwise we set it to null? 

I.e. if blockheight is null, block hash will be null also?